src/proposal.mligo

```diff
@@ -341,6 +363,68 @@ let delete_proposal
     Set.remove (start_date, proposal_key) store.proposal_key_list_sort_by_date
   }
 
+[@inline]
+let fraction_to_quorum_fraction(n, d : nat * nat): unsigned_quorum_fraction
+  = { numerator = (n * quorum_denominator) / d }
+
+[@inline]
+let to_signed(n : unsigned_quorum_fraction): quorum_fraction
+  = { numerator = int(n.numerator) }
+
+[@inline]
+let to_unsigned(n : quorum_fraction): unsigned_quorum_fraction
+  = { numerator = match is_nat(n.numerator) with
+              | Some n -> n
+              | None -> (failwith("BAD_UNSIGNED_CONVERSION"):nat)
+    }
+
+let update_quorum(store, config : storage * config): storage =
+  let current_period = get_current_period_num(store.start_time, config.voting_period) in
+  let current_cycle = period_to_cycle(current_period) in
+  if store.quorum_threshold_at_cycle.last_updated_cycle = current_cycle
+    then store // Quorum has been updated in this period, so no change is required.
+    else
+      if current_cycle > store.quorum_threshold_at_cycle.last_updated_cycle
+          then
+              let previous_staked = store.quorum_threshold_at_cycle.staked in
+              let previous_participation = to_signed(fraction_to_quorum_fraction(previous_staked, config.governance_total_supply)) in
+              let old_quorum = to_signed(store.quorum_threshold_at_cycle.quorum_threshold) in
+              let quorum_change = to_signed(config.quorum_change) in
+              let possible_new_quorum =
+                // old_quorum + (previous_participation - old_quorum) * quorum_change
+                fadd(old_quorum, fmul(quorum_change, fsub(previous_participation, old_quorum))) in
+                //old_quorum + fmul(quorum_change, previous_participation - old_quorum) in
```

Comment:
This line looks like it was left behind by accident.

Reply:
Removed